### DISCUSSION TITLE
Is it possible to get the username of the authenticated user?

### DISCUSSION THREAD

**Discussion by apastel** [sigma67/ytmusicapi#543] [Q&A] [answered]
Using the OAuth 2.0 flow, I believe this requires an identity token instead of an access token, although sometimes access tokens contain claims about the user's identity. The access tokens returned using the Google oauth client don't seem to be a JWT so I don't know how to decode them. Wondering if this is possible so that the user of my application can see who they are logged in as.

**Comment by sigma67** [comment, marked as answer]
The `account_menu` endpoint returns `accountName`, `channelHandle` and `accountPhoto`, if that's what you're looking for. Should be possible to implement without much effort if you need it

`https://music.youtube.com/youtubei/v1/account/account_menu`

**Reply by apastel** [reply]
Cool, good to know. Maybe this can be something to get my feet wet with contributing to the project if that would be of interest.

**Reply by sigma67** [reply]
Yes, for sure. Feel free to submit a PR

**Reply by sigma67** [reply]
Function could be called `get_account` in the `LibraryMixin`
